metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -163,10 +75,20 @@ func (sink *StackdriverSink) Stop() {
 func (sink *StackdriverSink) processMetrics(metricValues map[string]core.MetricValue,
 	timestamp time.Time, labels map[string]string, createTime time.Time) []*sd_api.TimeSeries {
 	timeseries := make([]*sd_api.TimeSeries, 0)
-	for name, value := range metricValues {
-		ts := sink.TranslateMetric(timestamp, labels, name, value, createTime)
-		if ts != nil {
-			timeseries = append(timeseries, ts)
+	if useOldResourceModel {
```

Comment:
I prefer not doing the refactoring in this PR. After talking to @fgrzadkowski I think it's not that obvious how the code should look like. I've written some example code refactored to a map, which still doesn't look ideal - feel free to talk to me offline and discuss which form is better. In general, I think this is a separate discussion and doesn't belong in this PR.

Reply:
Since @fgrzadkowski approves, I'm fine with duplication

However, personally I think it's better not to write code, rather than write it and then rewrite once again

Reply:
I'd prefer not to introduce such refactoring. First it's not obvious for me, it'll actually help with readability. Secondly, I'd like this PR to be minimal. We can refactor later.